src/formatters/deployResultFormatter.ts

```diff
@@ -124,10 +125,23 @@ export class DeployResultFormatter extends ResultFormatter {
   }
 
   protected displayFailures(): void {
-    if (this.hasStatus(RequestStatus.Failed) && this.fileResponses?.length) {
-      const failures = this.fileResponses.filter((f) => f.state === 'Failed');
-      this.sortFileResponses(failures);
-      this.asRelativePaths(failures);
+    if (this.hasStatus(RequestStatus.Failed)) {
+      const failures: Array<FileResponse | DeployMessage> = [];
+
+      if (this.fileResponses?.length) {
+        const fileResponses = this.fileResponses.filter((f) => f.state === 'Failed');
+        this.sortFileResponses(fileResponses);
+        this.asRelativePaths(fileResponses);
+        failures.push(...fileResponses);
+      }
+
+      if (this.result?.response?.details?.componentFailures) {
```

Comment:
aren't componentFailures and fileResponses going to include some of the same errors, and so this would show duplicates?

Reply:
ugh. dupes 🤦